routers/api/v2/users.go

```diff
@@ -280,6 +281,97 @@ func (r *apiV2Router) RemoveFromTeam(ctx *gin.Context) {
+	var req struct {
+		Password string `form:"password"`
+	}
+	_ = ctx.Bind(&req)
+	if len(req.Password) == 0 {
+		r.logger.Debug("password not specified", zap.Int("password length", len(req.Password)))
+		models.SendAPIError(ctx, http.StatusBadRequest, "request must include the new password")
+		return
+	}
+
+	user, err := r.getUserCtxAware(ctx, ctx.Param("id"))
+	if err != nil {
+		switch errors.Cause(err) {
+		case v2.ErrInvalidToken:
+			r.logger.Debug("invalid token", zap.Error(err))
+			r.HandleUnauthorized(ctx)
+		case v2.ErrInvalidTokenType:
+			r.logger.Debug("invalid token type", zap.Error(err))
+			models.SendAPIError(ctx, http.StatusBadRequest, "provided token is of invalid type for the requested operation")
+		case services.ErrInvalidID:
+			r.logger.Debug("invalid user id", zap.Error(err))
+			models.SendAPIError(ctx, http.StatusBadRequest, "invalid user id")
+		case services.ErrNotFound:
+			r.logger.Debug("user not found", zap.Error(err))
+			models.SendAPIError(ctx, http.StatusNotFound, "user not found")
+		default:
+			r.logger.Error("could not fetch user", zap.Error(err))
+			models.SendAPIError(ctx, http.StatusInternalServerError, "something went wrong")
+		}
+		return
+	}
+
+	err = r.userService.ResetPasswordForUserWithIDAndEmail(ctx, user.ID.Hex(), user.Email, req.Password)
+	if err != nil {
+		switch errors.Cause(err) {
+		case services.ErrInvalidID:
+			r.logger.Debug("invalid user id", zap.Error(err))
+			models.SendAPIError(ctx, http.StatusBadRequest, "invalid user id")
+		case services.ErrNotFound:
+			r.logger.Debug("user not found", zap.Error(err))
+			models.SendAPIError(ctx, http.StatusNotFound, "user does not exist")
+		default:
+			r.logger.Error("could not update user", zap.Error(err))
+			models.SendAPIError(ctx, http.StatusInternalServerError, "something went wrong")
+		}
+	}
+
+	ctx.Status(http.StatusOK)
+}
+
+// GET: /api/v2/users/(:id|me)/password/resetEmail
+// Response:
+// Headers:  Authorization -> token
+func (r *apiV2Router) GetPasswordResetEmail(ctx *gin.Context) {
+	user, err := r.getUserCtxAware(ctx, ctx.Param("id"))
+	if err != nil {
+		switch errors.Cause(err) {
+		case v2.ErrInvalidToken:
+			r.logger.Debug("invalid token", zap.Error(err))
+			r.HandleUnauthorized(ctx)
+		case v2.ErrInvalidTokenType:
+			r.logger.Debug("invalid token type", zap.Error(err))
+			models.SendAPIError(ctx, http.StatusBadRequest, "provided token is of invalid type for the requested operation")
+		case services.ErrInvalidID:
+			r.logger.Debug("invalid user id", zap.Error(err))
+			models.SendAPIError(ctx, http.StatusBadRequest, "invalid user id")
+		case services.ErrNotFound:
+			r.logger.Debug("user not found", zap.Error(err))
+			models.SendAPIError(ctx, http.StatusNotFound, "user not found")
+		default:
+			r.logger.Error("could not fetch user", zap.Error(err))
+			models.SendAPIError(ctx, http.StatusInternalServerError, "something went wrong")
+		}
+		return
+	}
+
+	err = r.emailService.SendPasswordResetEmail(*user)
```

Comment:
The SendGrid client uses V1 of the auth system, so `SendPasswordResetEmail` will need updating as well. Maybe just create a TODO for now since that would break API V1.

Reply:
Ah true, I assumed the `SendPasswordResetEmail` would need to be updated anyway when we start working on the `hs_notify` service

Reply:
Hmm true, I think just add a TODO with a GH issue and that'll be fine.